### PR TITLE
Feature/register login api

### PR DESCRIPTION
**Pull request by Foteno**
Added register, login, added hashing of passwords, added exceptions

### CODE REVIEW

File: Backend/src/main/java/lt/vu/ads/service/courier/CourierServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public CourierLoggedInView login(CourierLoginView loginView) {
+        Courier courier = courierRepository.findByEmail(loginView.getEmail());
+
+        if (courier == null) {
+            throw new CustomException("Tokio vartotojo nėra");
+        }
+        if (!passwordEncoder.matches(loginView.getPassword(), courier.getPassword())) {
+            throw new WrongPasswordException("Neteisingas slaptažodis");
```
Comment (Tomas-Per) [inline Backend/src/main/java/lt/vu/ads/service/courier/CourierServiceImpl.java]: Messages should be in English preferably.

File: Backend/src/main/java/lt/vu/ads/security/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import lombok.RequiredArgsConstructor;
+
+@Configuration
+@RequiredArgsConstructor
+public class SecurityConfig {
+
+    @Bean
+    public BCryptPasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
```
Comment (Tomas-Per) [inline Backend/src/main/java/lt/vu/ads/security/SecurityConfig.java]: Maybe change this to `return new BCryptPasswordEncoder(12);` (default strength is 10) for better security

File: Backend/src/main/java/lt/vu/ads/service/user/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public UserLoggedInView login(UserLoginView loginView) {
+        User user = userRepository.findByEmailAndPassword(loginView.getEmail(), loginView.getPassword());
+
+        if (user == null) {
+            throw new CustomException("Tokio vartotojo nėra");
+        }
+        if (!passwordEncoder.matches(loginView.getPassword(), user.getPassword())) {
+            throw new WrongPasswordException("Neteisingas slaptažodis");
```
Comment (Tomas-Per) [inline Backend/src/main/java/lt/vu/ads/service/user/UserServiceImpl.java]: English.

**Review by Tomas-Per — CHANGES_REQUESTED** [review]

File: Backend/src/main/java/lt/vu/ads/service/courier/CourierServiceImpl.java (hunk 1 above)
Comment (Foteno) [reply Backend/src/main/java/lt/vu/ads/service/courier/CourierServiceImpl.java]: agreed

File: Backend/src/main/java/lt/vu/ads/security/SecurityConfig.java (hunk 2 above)
Comment (Foteno) [reply Backend/src/main/java/lt/vu/ads/security/SecurityConfig.java]: agreed

**Review by Tomas-Per — APPROVED** [review]

**Comment by Filari02** [comment]
seems good

**Review by Ugniusst — APPROVED** [review]

**Review by Filari02 — APPROVED** [review]

**Review by aule2000 — APPROVED** [review]
Good

**Pull request merged**